### PR TITLE
Add support for http4k Bolt adapter

### PR DESCRIPTION
**Pull request by daviddenton**
Implements #584 

### Category (place an `x` in each of the `[ ]`)

* [ ] **bolt** (Bolt for Java)
* [X] **bolt-{sub modules}** (Bolt for Java - optional modules)
* [ ] **slack-api-client** (Slack API Clients)
* [ ] **slack-api-model** (Slack API Data Models)
* [ ] **slack-api-*-kotlin-extension** (Kotlin Extensions for Slack API Clients)
* [ ] **slack-app-backend** (The primitive layer of Bolt for Java)

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to the those rules.

### CODE REVIEW

**Comment by seratch** [comment]
@daviddenton Thank you very much for submitting this PR! The adapter for http4k is a great addition to this SDK.

To move this forward, could you check the following?

* My updates for the tests - https://github.com/http4k/java-slack-sdk/pull/1
* Signing the CLA is required for merging this PR 🙇

**Comment by daviddenton** [comment]
Done and done. 😃

File: bolt-http4k/src/main/java/test_locally/Http4kSlackApp.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        this.app = app;
+        requestParser = new SlackRequestParser(app.config());
+    }
+
+    @Override
+    public Response invoke(Request request) {
+        try {
+            return toHttp4kResponse(app.run(toSlackRequest(request)));
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (seratch) [inline bolt-http4k/src/main/java/test_locally/Http4kSlackApp.java:35]: Ah, I overlooked this. Can you add 

```java
    private static final Logger LOGGER = LoggerFactory.getLogger(Http4kSlackApp.class);
```

and then, update this line?

```suggestion
            LOGGER.error(e.getMessage(), e);
```

File: bolt-http4k/src/main/java/test_locally/Http4kSlackAppServer.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        if (server != null) server.stop();
+        return this;
+    }
+
+    @Override
+    public void block() {
+        try {
+            Thread.currentThread().join();
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```
Comment (seratch) [inline bolt-http4k/src/main/java/test_locally/Http4kSlackAppServer.java:37]: Just out of curiosity, does http4k handle this runtime exception properly? or should developers be careful about shutting down an app? As you're the creator of http4k, I believe this is the best way but I just wanted to understand what this method is.

Comment (daviddenton) [reply bolt-http4k/src/main/java/test_locally/Http4kSlackAppServer.java]: That exception won't actually ever occur in the normal running of an App. It's actually only ever used at the top level (it's part of the outermost Server API), and is a bit of an integration niggle due to checked/unchecked exceptions.

File: bolt-http4k/src/main/java/com/slack/api/bolt/http4k/Http4kSlackApp.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package test_locally;
```
Comment (seratch) [inline bolt-http4k/src/main/java/com/slack/api/bolt/http4k/Http4kSlackApp.java]: @daviddenton This test package name is a convention in this project. Can you revert this change?

**Review by seratch — CHANGES_REQUESTED** [review]

File: bolt-http4k/src/main/java/com/slack/api/bolt/http4k/Http4kSlackApp.java (hunk 3 above)
Comment (seratch) [reply bolt-http4k/src/main/java/com/slack/api/bolt/http4k/Http4kSlackApp.java]: ah, no. you're right. thanks!

File: bolt-http4k/src/main/java/test_locally/Http4kSlackAppServer.java (hunk 2 above)
Comment (seratch) [reply bolt-http4k/src/main/java/test_locally/Http4kSlackAppServer.java]: That makes sense. Thanks!

**Review by seratch — APPROVED** [review]
Thanks for promptly updating the branch. I'm going to merge this after a successful CI build. We'll release a new minor version including this new module soon (probably by the end of this month).

**Comment by daviddenton** [comment]
@seratch Regarding the package change - I've changed just the prod code to be in the original package, not the tests. Could you confirm that's ok?

**Comment by seratch** [comment]
@daviddenton Yes, that was my bad. Thanks a lot for detecting it 👍

**Comment by daviddenton** [comment]
@seratch Unless I'm very much mistaken, travis seems to be failing on something unrelated. Any idea?

**Comment by seratch** [comment]
I don't see any issues, so I reran the build. Somehow, it succeded as with your first commit. I will check this PR again later and merge it.

**Comment by daviddenton** [comment]
Is there anything else that needs to happen to make sure the artifact gets published properly to the maven repo?

File: README.md [hunk 4]
```diff
@@ -68,6 +68,7 @@ All released versions are available on the Maven Central repositories. The lates
 |[**com.slack.api:bolt-google-cloud-functions**](https://search.maven.org/search?q=g:com.slack.api%20AND%20a:bolt-google-cloud-functions)|This module offers a handy way to run Bolt apps on [Google Cloud Functions](https://cloud.google.com/functions).|
 |[**com.slack.api:bolt-micronaut**](https://search.maven.org/search?q=g:com.slack.api%20AND%20a:bolt-micronaut)|This is an adapter for [Micronaut](https://micronaut.io/) to run Bolt apps on top of it.|
 |[**com.slack.api:bolt-helidon**](https://search.maven.org/search?q=g:com.slack.api%20AND%20a:bolt-helidon)|This is an adapter for [Helidon SE](https://helidon.io/docs/latest/) to run Bolt apps on top of it.|
+|[**com.slack.api:bolt-http4k**](https://search.maven.org/search?q=g:com.slack.api%20AND%20a:bolt-http4k)|This is an adapter for [http4k](https://http4k.org/) to run Bolt apps on top of any of the multiple server backends that the library supports.|
```
Comment (seratch) [inline README.md:71]:
```
|[**com.slack.api:bolt-http4k**](https://search.maven.org/search?q=g:com.slack.api%20AND%20a:bolt-http4k)|This is an adapter for [http4k](https://http4k.org/) to run Bolt apps on top of any of the multiple server backends that the library supports.|
```
This is great! However, as we still need a bit more time to release v1.3, let me revert this so far. I will surely include this exact sentence in README and the website when we release v1.3.
```suggestion
```

**Comment by seratch** [comment]
>Is there anything else that needs to happen to make sure the artifact gets published properly to the maven repo?
>I added the module, by wasn't sure if there was another change to be made.

I manually verified the module's build settings do not have any issues. Here is the uploaded snapshot version: https://oss.sonatype.org/content/repositories/snapshots/com/slack/api/bolt-http4k/

**Comment by seratch** [comment]
Thanks for your contribution! The changes have been merged in https://github.com/slackapi/java-slack-sdk/commit/5581fe7aa952680312cad93f62cb1b25fd5b3cd2
